refaster-support/src/test/java/tech/picnic/errorprone/refaster/AnnotatedCompositeCodeTransformerTest.java

```diff
@@ -0,0 +1,203 @@
+package tech.picnic.errorprone.refaster;
+
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+import static com.google.errorprone.BugPattern.SeverityLevel.SUGGESTION;
+import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.notNull;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.auto.value.AutoAnnotation;
+import com.google.common.collect.ImmutableClassToInstanceMap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.CodeTransformer;
+import com.google.errorprone.DescriptionListener;
+import com.google.errorprone.ErrorProneOptions;
+import com.google.errorprone.fixes.Fix;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.util.TreePath;
+import com.sun.tools.javac.util.Context;
+import com.sun.tools.javac.util.JCDiagnostic.DiagnosticPosition;
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import tech.picnic.errorprone.refaster.annotation.OnlineDocumentation;
+import tech.picnic.errorprone.refaster.annotation.Severity;
+
+// XXX: Test the `ErrorProneOptions`-based severity override logic. (Right now that logic is tested
+// through `RefasterTest`, but ideally it is covered by tests in this class, closer to the code that
+// implements the relevant logic.)
+final class AnnotatedCompositeCodeTransformerTest {
+  private static final DiagnosticPosition DUMMY_POSITION = mock(DiagnosticPosition.class);
+  private static final Fix DUMMY_FIX = mock(Fix.class);
+  private static final TreePath DUMMY_PATH = mock(TreePath.class);
+  private static final String DEFAULT_PACKAGE = "";
+  private static final String CUSTOM_PACKAGE = "com.example";
+  private static final String SIMPLE_CLASS_NAME = "MyRefasterRule";
+
+  private static Stream<Arguments> applyTestCases() {
+    /* { context, packageName, ruleName compositeAnnotations, delegateAnnotations, expectedDescription } */
+    return Stream.of(
+        arguments(
+            context(),
```

Comment:
We pass `context()` for every test case, would it make sense to move this to the test method and instantiate it there?

Reply:
As-is: yes. But when we pick up the `XXX` this parameter is important. Let me move the comment there.

Reply:
Done.